core/iam_authenticator.go

```diff
@@ -32,9 +32,12 @@ const (
 	REQUEST_TOKEN_RESPONSE_TYPE = "cloud_iam"
 )
 
-// IamAuthenticator : This authenticator will automatically fetch an access token for the
-// configured apikey.  Outbound REST requests invoked by the BaseService are then authenticated
-// by adding a Bearer-type Authorization header containing the access token.
+// IamAuthenticator utilizes an apikey, or ClientId and ClientSecret
+// pair to obtain a suitable bearer token, and adds it to requests.
+//
+// The bearer token will be sent as an Authorization header in the form:
+//
+// 		Authorization: Bearer <bearer-token>
 type IamAuthenticator struct {
 
 	// [Required] The apikey used to fetch the access token from the IAM token server.
```

Comment:
```suggestion
	// The apikey used to fetch the bearer token from the IAM token server [required].
```

Reply:
the comment for URL should be changed to:
```
// The URL representing the IAM token server's endpoint [optional].
// If not specified, a suitable default value will be used.
```

Reply:
also, move the [Optional] tags to the end of the descriptions for the rest of the fields to be consistent with previous struct fields.